src/solvespace.cpp

```diff
@@ -331,7 +331,13 @@ const char *SolveSpaceUI::UnitName() {
 
 std::string SolveSpaceUI::MmToString(double v) {
     v /= MmPerUnit();
-    return ssprintf("%.*f", UnitDigitsAfterDecimal(), v);
+    int digits = UnitDigitsAfterDecimal();
+    double minimum = 0.5 * pow(10,-digits);
```

Comment:
How about

```suggestion
    if(v < pow(10, -digits)) {
        return ssprintf("%g", v);
    }
    return ssprintf("%.*f", digits, v);
```

In this way if `0.000` would have been displayed (or any rounding in the most significant digit) the value is instead displayed in scientific notation (e.g, `7.23e-08`).

Editing becomes very inconvenient (since the edit box shows the %g format, but a normal number is expected) and the used is forced to
- live with it
- increase the value
- Increase `digits after decimal point to show` (display more significant digits)

Reply:
I don't really want scientific notation to start appearing, but the input boxes seem to support it. Also, I used the 0.5 because 0.006 would display as 0.01 for 2 digits while 0.005 would be 0.00. Is this PR better than what we have? Does it solve the problem of showing 0 when the value is not 0? And given recent changes, is it correct?  ;-)

Reply:
I understood your logic behind 0.5 (the rounding).
My proposal of just `pow(10, -digits)` is because with the 0.5 (and the current master logic) if one types:
0.0006234
one would get
0.001
displayed. And the same when editing. So they would loose the extra significant digits _visually_ while interrnally they are still there. This is misleading.

With `pow(10, -digits)` they would get the scientific notation with anything that does not have a non-zero digit in the first N (whatever N is set to).

As for the overall logic:
Yours:
- is nice since it shows the complete number always.
- It is "confusing" because it ignores the number of significant digits the user has set when a number is too small.

Mine:
- is nice, because it respects the users choice of number of significant digits and does something obviously "weird" (the scientific notation) when it can not. This would (hopefully) force the user to think what is going on.
- It is annoying, because not all edit fields support scientific notation and the user will have to re-type the value. For example the ones that use `Expr::From` do not (e.g.  `cutter radius offset` https://github.com/solvespace/solvespace/blob/bd41485ebecce360baa729567273714caac1df6c/src/confscreen.cpp#L497)

So which one is better? Is either of them is better than the current situation? I am not sure...

> And given recent changes, is it correct? ;-)

What recent changes?

Reply:
Mine and your changes that needed to be fixed after merging ;-)

Both of these solve the issue of displaying zero when a value is not zero, so I think it's an improvement.